### PR TITLE
Load examples manifest via HTTP

### PR DESCRIPTION
**Pull request by jprochazk**
### What

* Closes https://github.com/rerun-io/rerun/issues/4278
* Closes https://github.com/rerun-io/rerun/issues/3190 - it's now possible to patch the manifest for a previously released version, starting with the next release.

- [x] Stop including `examples_manifest.json` as part of the binary in favor of downloading it from `app.rerun.io`
- [x] Move `re_viewer/build.rs` logic to a separate binary crate
- [x] Native: Manifest URL override via a runtime env var
- [x] Web: Manifest URL override via query param (`manifest_url`)
- [x] CI: Build and upload `examples_manifest.json` together with every build of the web app for:
  - `/commit`
  - `/pr`
  - `/version/nightly`
  - `/version/{tag}`
  - `/prerelease`
  - `/latest`
- [x] PR template: Link to app using both nightly and PR examples manifest

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Full build: [app.rerun.io](https://app.rerun.io/pr/4391/index.html)
  * Partial build: [app.rerun.io](https://app.rerun.io/pr/4391/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json) - Useful for quick testing when changes do not affect examples in any way
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4391)
- [Docs preview](https://rerun.io/preview/cf3b24badc3c29a75442aa2eb27ba7eeadcdb7ac/docs) 
- [Examples preview](https://rerun.io/preview/cf3b24badc3c29a75442aa2eb27ba7eeadcdb7ac/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Review by emilk — APPROVED** [review]
Looks good overall, but I don't understand why there are two links in the new PR description template

File: crates/re_viewer/src/ui/welcome_screen/example_page.rs [hunk 1]
```diff
@@ -221,15 +291,15 @@ impl ExamplePage {
                             });
                     });
 
-                self.examples.iter().for_each(|example| {
+                for example in examples {
```
[the remark this answers is not in the document]
Comment (jprochazk) [reply crates/re_viewer/src/ui/welcome_screen/example_page.rs:294]: I'm not sure what I changed, but at some point while I was working on this, the lint finally popped up which prompted me to change this

File: web_viewer/index.html [hunk 2]
```diff
@@ -351,9 +351,9 @@
 
         check_for_panic();
 
-        let url = new URLSearchParams(window.location.search).get("url");
+        let query = new URLSearchParams(window.location.search);
         handle
-          .start("the_canvas_id", url)
+          .start("the_canvas_id", query.get("url"), query.get("manifest_url"))
```
[the remark this answers is not in the document]
Comment (jprochazk) [reply web_viewer/index.html:356]: It is indeed, the arguments on `handle.start` are meant more as overrides (so we can expose them from the npm package, for example), but I'm not sure why we fetch these twice. I'll remove this.

**Comment by jprochazk** [comment]
I pulled in an unrelated change, prettier v3 expects lowercase `doctype`, but prettier v2 expects it uppercased. The vscode extension has yet to update to prettier v3, so the solution for now is to downgrade the version in `pixi.toml` to v2.

[prettier/prettier#15476](https://www.github.com/prettier/prettier/issues/15476)

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -15,7 +15,9 @@ To get an auto-generated PR description you can put "copilot:summary" or "copilo
 ### Checklist
 * [ ] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
 * [ ] I've included a screenshot or gif (if applicable)
-* [ ] I have tested [app.rerun.io](https://app.rerun.io/pr/{{ pr.number }}) (if applicable)
+* [ ] I have tested the web demo (if applicable):
+  * [Nightly](https://app.rerun.io/pr/{{ pr.number }}/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
+  * [PR](https://app.rerun.io/pr/{{ pr.number }}/index.html)
```
[the remark this answers is not in the document]
Comment (jprochazk) [reply .github/pull_request_template.md]: I tried to add some more information here

**Pull request merged**